test/js_test.go

```diff
@@ -3841,30 +3832,24 @@ func testJetStream_ClusterReconnectDurableQueueSubscriber(t *testing.T, subject
 	}
 
 	// Check for persisted messages, this could fail a few times.
-	var stream *nats.StreamInfo
-	timeout := time.Now().Add(5 * time.Second)
-	for time.Now().Before(timeout) {
-		stream, err = js.StreamInfo(subject)
-		if err == nats.ErrTimeout {
-			time.Sleep(100 * time.Millisecond)
-			continue
-		} else if err != nil {
-			t.Fatalf("Unexpected error: %v", err)
-		}
-		break
+	stream, err := js.StreamInfo(subject,
+		nats.MaxRetries(10), nats.RetryBackoff(100*time.Millisecond))
```

Comment:
I think to replace this test it would be something like:

```go
ctx, err := context.WithTimeout(context.Background(), 5 * time.Second)
stream, err := js.StreamInfo(subject, nats.Context(ctx), nats.MaxRetries(-1), nats.RetryBackoff(100*time.Millisecond))
```

Would that work?

Reply:
Ah, do we want to allow infinite retries? With `nats.MaxRetries(-1)`?

Reply:
I think -1 would be fine to support, since the context would ensure that it gets canceled, also iirc a default context based on the default js.wait 5seconds default is getting created so would not block.